app/v2/utils/MessageFormat.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.utils
+
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.data.EitherT
+import v2.controllers.ErrorTranslator
+import v2.models.errors.MessageFormatError
+import v2.models.errors.PresentationError
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+object MessageFormat extends ErrorTranslator {
+
+  def convertSourceToString(
+    source: Source[ByteString, _]
+  )(implicit ec: ExecutionContext, mat: Materializer): EitherT[Future, MessageFormatError, String] = EitherT {
+    source
+      .reduce(_ ++ _)
+      .map(_.utf8String)
+      .runWith(Sink.head[String])
+      .map(Right[MessageFormatError, String])
+      .recover {
+        case NonFatal(ex) => Left[MessageFormatError, String](MessageFormatError.UnexpectedError(Some(ex)))
```

Comment:
Not sure about the name, code's fine, just MessageFormat seems a bit weird for something that's essentially a tool to realise or force a stream to complete into a string. StreamComplete? CompleteSource? ResolveSource?

Reply:
StreamingUtils?